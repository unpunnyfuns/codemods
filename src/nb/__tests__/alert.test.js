import { describe, expect, it } from 'vitest'
import { testTransform } from '../../__tests__/test-helper.js'

describe('nb/alert', () => {
  it('migrates Alert component', () => {
    const output = testTransform('nb/alert', 'nb/alert', 'tsx')
    expect(output).toMatchSnapshot()
  })
})
