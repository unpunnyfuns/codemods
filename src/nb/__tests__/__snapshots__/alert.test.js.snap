// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/alert > migrates Alert component 1`] = `
"import { Alert } from '@hb-frontend/app/src/components/nordlys/Alert';
import { View, StyleSheet } from 'react-native';
import { space } from '@hb-frontend/nordlys';

export function Example() {
  return (
    <>
      {/* Basic Alert with compound components */}
      <Alert
        status="success"
        title='Success'
        description='Your changes have been saved' />
      {/* Alert with expression */}
      <Alert status="error" title='Error' description={errorMessage} />
      {/* Alert with testID */}
      <Alert
        status="info"
        testID="info-alert"
        title='Information'
        description='Please review the details below' />
      {/* Alert with layout props */}
      <View style={styles.alert3}>
        <Alert status="warning" description='This action cannot be undone' />
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  alert3: {
    marginTop: space.md,
    marginBottom: space.lg
  }
});
"
`;
